subprojects/snapshots/src/test/groovy/org/gradle/internal/fingerprint/impl/FingerprintCompareStrategyTest.groovy

```diff
@@ -266,6 +266,54 @@ class FingerprintCompareStrategyTest extends Specification {
         NORMALIZED | false
     }
 
+    @Unroll
+    def "mix file contents with same name (#strategy, include added: #shouldIncludeAdded)"() {
+        def changes = changes(strategy, shouldIncludeAdded,
+            ["a/absolute": fingerprint("absolute", 2),
+             "a/relative": fingerprint("a/relative", 3),
+
+             "b/absolute": fingerprint("absolute", 0),
+             "b/relative": fingerprint("b/relative", 1)],
+
+            ["a/absolute": fingerprint("absolute", 0),
+             "a/relative": fingerprint("a/relative", 1),
+
+             "b/absolute": fingerprint("absolute", 2),
+             "b/relative": fingerprint("b/relative", 3)]
+        )
+        expect:
+        changes*.toString().toSet() == results*.toString().toSet()
+
+        where:
+        strategy   | shouldIncludeAdded | results
+        NORMALIZED | true         | [modified("a/relative"), modified("b/relative"), modified("a/absolute"), modified("b/absolute")]
+        NORMALIZED | false        | [modified("a/relative"), modified("b/relative"), modified("a/absolute"), modified("b/absolute")]
+    }
+
+    @Unroll
+    def "change file content to other content with same name (#strategy, include added: #shouldIncludeAdded)"() {
+        def changes = changes(strategy, shouldIncludeAdded,
+            ["a/absolute": fingerprint("absolute", 2),
+             "a/relative": fingerprint("a/relative", 3),
+
+             "b/absolute": fingerprint("absolute", 2),    // unchanged
+             "b/relative": fingerprint("b/relative", 3)], // unchanged
+
+            ["a/absolute": fingerprint("absolute", 0),
+             "a/relative": fingerprint("a/relative", 1),
+
+             "b/absolute": fingerprint("absolute", 2),
+             "b/relative": fingerprint("b/relative", 3)]
+        )
+        expect:
+        changes*.toString().toSet() == results*.toString().toSet()
+
+        where:
+        strategy   | shouldIncludeAdded | results
+        NORMALIZED | true         | [modified("a/relative"), modified("a/absolute")]
+        NORMALIZED | false        | [modified("a/relative"), modified("a/absolute")]
```

Comment:
Nit: please reformat "table".

Reply:
yeah, was the first thing I noticed after push, but since I've already sent the links to the commits, only fixed it locally.